domain/modelagent/state/modelstate.go

```diff
@@ -185,33 +455,48 @@ WHERE  mav.machine_uuid = $machineAgentVersionInfo.machine_uuid
 
 // GetUnitTargetAgentVersion returns the target agent version for the specified unit.
 // The following error types can be expected:
-// - [modelagenterrors.AgentVersionNotFound] - when the agent version does not exist.
+// - [applicationerrors.UnitNotFound] when the unit does not exist.
+// - [modelagenterrors.AgentVersionNotFound] when the agent version does not exist.
 func (st *State) GetUnitTargetAgentVersion(ctx context.Context, uuid coreunit.UUID) (coreagentbinary.Version, error) {
 	db, err := st.DB()
 	if err != nil {
 		return coreagentbinary.Version{}, errors.Capture(err)
 	}
 
-	info := unitAgentVersionInfo{UnitUUID: uuid}
+	info := unitTargetAgentVersionInfo{}
+	unitUUID := unitUUIDRef{
+		UUID: uuid,
+	}
 
 	stmt, err := st.Prepare(`
-SELECT av.target_version AS &unitAgentVersionInfo.target_version,
-       a.name AS &unitAgentVersionInfo.architecture_name
-FROM   agent_version AS av,
-       unit_agent_version AS uav
-JOIN   architecture AS a ON uav.architecture_id = a.id
-WHERE  uav.unit_uuid = $unitAgentVersionInfo.unit_uuid
-`, info)
+SELECT &unitTargetAgentVersionInfo.*
+FROM v_unit_target_agent_version
+WHERE unit_uuid = $unitUUIDRef.unit_uuid
+`, info, unitUUID)
 	if err != nil {
 		return coreagentbinary.Version{}, errors.Capture(err)
 	}
 
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		err := tx.Query(ctx, stmt, info).Get(&info)
+		err := st.checkUnitNotDead(ctx, tx, uuid)
+		if errors.Is(err, applicationerrors.UnitNotFound) {
+			return errors.Errorf(
+				"unit %q does not exist", uuid,
+			).Add(applicationerrors.UnitNotFound)
+		} else if err != nil && !errors.Is(err, applicationerrors.UnitIsDead) {
+			return errors.Errorf(
+				"checking if unit %q exists: %w", uuid, err,
+			)
+		}
```

Comment:
Same here and if you wanted to check this, why not copy `checkUnitNotDead` and `checkMachineNotDead` to make `checkUnitExists` and `checkMachineExists`.

Reply:
I should have done this just got lazy.

Reply:
Done & Done